### PR TITLE
Syn 163 show statistics of a trace

### PR DESCRIPTION
**Pull request by MathiasBoi6**
## Describe your changes
Did some stats, fixed some issues, especially with tablePreview. 
Still a warning when configuration is unselected, too much work to be worth fixing.
!There is an issue with simulation. While using an empty seed, the resulting traces are always the same. Len was doing something with the random seed, so I have not fixed it here.

## Checklist before requesting a review
- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by Volume999** [comment]
Once the Mass generation of traces is merged to main, you will have to change a lot of things here, I reckon? So merging this would not really be a wise thing right now I think

File: packages/synthchron-tauri/src/components/PostProcessing.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ import {
 
 import { CustomAppBar } from './CustomAppBar'
 
-function XESlogToString(log: XESLog) {
+export function XESlogToString(log: XESLog) {
```
Comment (Volume999) [inline packages/synthchron-tauri/src/components/PostProcessing.tsx:20]: Pascal Case

File: packages/synthchron-tauri/src/components/editor/bottomDrawer/AnalysisPanel.tsx [hunk 2]
```diff
@@ -103,90 +91,97 @@ export const AnalysisPanel: React.FC<AnalysisPanelProps> = () => {
...
+        }}
+      >
+        Export as XES with fully serialized Model
+      </Button>
+    </Stack>
+  )
+
+  const aggregateOfLog = TransformToAggregate(result.log)
+
+  const DoughnotData = AggregateToData(aggregateOfLog, chartType.Doughnot)
```
Comment (Volume999) [inline packages/synthchron-tauri/src/components/editor/bottomDrawer/AnalysisPanel.tsx:179]: Typo + camel case

File: packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
+import { faker } from '@faker-js/faker'
+
+import { XESLog } from '@synthchron/xes'
+
+type logAggregate = Map<string, number> //Map<key, value>
```
Comment (Volume999) [inline packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx:5]: Pretty sure types are pascal case

File: packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+//Also bar and pie chart
+type doughnotData = {
```
Comment (Volume999) [inline packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx:8]: typo and pretty sure pascal case

File: packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  label: string
+  data: number[]
+  backgroundColor: string[]
+  hoverOffset: number
+}
+
+type defualtData = {
```
Comment (Volume999) [inline packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx:15]: typo

File: packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+type dataset = doughnotData | defualtData
+
+type chartData = {
+  labels: string[]
+  datasets: dataset[]
+}
+
+export enum chartType {
+  Doughnot,
```
Comment (Volume999) [inline packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx:30]: typo and I think enums are also pascal case?

File: packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+type defualtData = {
+  label: string
+  data: number[]
+  borderColor: string
+  backgroundColor: string
+}
+
+type dataset = doughnotData | defualtData
```
Comment (Volume999) [inline packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx:22]: Pascal case I think

File: packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+type chartData = {
```
Comment (Volume999) [inline packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx:24]: also Pascal Case

File: packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+  labels: string[]
+  datasets: dataset[]
+}
+
+export enum chartType {
+  Doughnot,
+  Other,
+}
+
+//REMOVE EXPORT
```
Comment (Volume999) [inline packages/synthchron-tauri/src/components/editor/bottomDrawer/analysisFunctions/aggregateCharts.tsx:34]: you didn't remove export

**Review by Volume999 — CHANGES_REQUESTED** [review]

**Review by Volume999 — CHANGES_REQUESTED** [review]
Ultimately, you will also have to merge it with the mass trace generation PR in order to start fully evaluating your PR

**Comment by MathiasBoi6** [comment]
Making a link to this image so i can reference it in my code :)
image

**Review by leny-mi — APPROVED** [review]
lgtm!

**Review by Volume999 — APPROVED** [review]

**Pull request merged**